service/src/main/java/org/ehrbase/service/PartyServiceImp.java

```diff
@@ -64,19 +70,16 @@ public UUID getCurrentUserId() {
     private UUID getOrCreateCurrentUserIdSync(CacheKey<String> key) {
 
         var existingUser = partyProxyRepository.findInternalUserId(key.getVal());
```

Comment:
If `::createInternalUser` is successfull, 
do we need to do `::findInternalUserId` again, 
or is this only needed in case of `DataIntegrityViolationException`?

Suggestion:
``` java
  return partyProxyRepository
          .findInternalUserId(key.getVal())
          .or(() -> {
              try {
                  return partyProxyRepository.createInternalUser(key.getVal());
              } catch (DataIntegrityViolationException ex) {
                  logger.info(ex.getMessage(), ex.getMessage());
                  return partyProxyRepository.findInternalUserId(key.getVal());
              }
          })
          .orElseThrow(() -> new InternalServerException("Cannot create User"));
```